app/services/orchestration/task.rb

```diff
@@ -24,11 +26,12 @@ def status?(symbol)
     end
 
     def to_s
-      "#{name}\t #{priority}\t #{status}\t #{action}"
+      tid = id.nil? ? '' : " (#{id})"
```

Comment:
woud it be better to rather have `(no id)` or something in case there's no id set? hopefully that would indicate, that task author should consider setting id

Reply:
Yeah, I actually plan to do DEPRECATED warning in case of missing id, that should be explicit enough I think.

Reply:
New version has `id` required now, deprecation warning raised when it's missing. I want to keep both ID and name here just to be clear about this.

Reply:
Can't we generate the id automatically by default, based on the task name, action class name, action method, action object id, priority, ... maybe using a hash?

This seems to be a lot of change effort in plugins just to detect duplicates.
Why doesn't something as simple as `[task1, task2, task3].uniq` work?
Or clearing the queue(s) before new tasks are added to make sure they are only added once?

Reply:
I'm not sure how many plugins extend the orchestration but if there are some, we should update them too so they pass id, otherwise it won't happen until we hard enforce it since all plugins would rely on defaulting to task name or action method. Since errors caused by duplicated tasks in queue are hard to debug, I'd prefer 500 unless that means updating all foreman plugins. If we talk about ~5, I think it's worth it